src/tigerbeetle/benchmark_load.zig

```diff
@@ -163,90 +169,96 @@ pub fn main(
 
     var benchmark = Benchmark{
         .io = &io,
-        .message_pool = &message_pool,
-        .client = &client,
-        .batch_accounts = batch_accounts,
+        .statsd = if (statsd_opt) |*statsd| statsd else null,
+        .random = random,
+        .timer = try std.time.Timer.start(),
+        .output = std.io.getStdOut().writer().any(),
+        .clients = clients.slice(),
+        .client_requests = client_requests,
+        .client_replies = client_replies,
+        .request_latency_histogram = request_latency_histogram,
+        .account_id_permutation = account_id_permutation,
+        .account_batch_size = cli_args.account_batch_size,
         .account_count = cli_args.account_count,
         .account_count_hot = cli_args.account_count_hot,
         .account_generator = account_generator,
         .account_generator_hot = account_generator_hot,
-        .flag_history = cli_args.flag_history,
-        .flag_imported = cli_args.flag_imported,
-        .account_index = 0,
-        .query_count = cli_args.query_count,
-        .query_index = 0,
-        .account_id_permutation = account_id_permutation,
-        .rng = rng,
-        .timer = try std.time.Timer.start(),
-        .batch_latency_histogram = batch_latency_histogram,
-        .query_latency_histogram = query_latency_histogram,
-        .batch_transfers = batch_transfers,
-        .batch_start_ns = 0,
+        .transfer_id_permutation = account_id_permutation,
+        .transfer_batch_size = cli_args.transfer_batch_size,
+        .transfer_batch_delay_us = cli_args.transfer_batch_delay_us,
         .transfer_count = cli_args.transfer_count,
         .transfer_hot_percent = cli_args.transfer_hot_percent,
         .transfer_pending = cli_args.transfer_pending,
-        .transfer_batch_size = cli_args.transfer_batch_size,
-        .transfer_batch_delay_us = cli_args.transfer_batch_delay_us,
+        .query_count = cli_args.query_count,
+        .flag_history = cli_args.flag_history,
+        .flag_imported = cli_args.flag_imported,
         .validate = cli_args.validate,
-        .batch_index = 0,
-        .transfers_sent = 0,
-        .transfer_index = 0,
-        .transfer_next_arrival_ns = 0,
-        .callback = null,
-        .done = false,
-        .statsd = if (statsd_opt) |*statsd| statsd else null,
         .print_batch_timings = cli_args.print_batch_timings,
-        .id_order = cli_args.id_order,
-        .batch_account_ids = batch_account_ids,
-        .batch_transfer_ids = batch_transfer_ids,
     };
 
-    benchmark.client.register(Benchmark.register_callback, @intCast(@intFromPtr(&benchmark)));
-    while (!benchmark.done) {
-        benchmark.client.tick();
+    benchmark.run(.register);
```

Comment:
If I am reading the code correctly, `run` is always followed by the exact same loop, so we perhaps can move the loop inside the run?